hedgehog/hedgehog.cabal

```diff
@@ -58,6 +58,7 @@ library
     , containers                      >= 0.4        && < 0.7
     , directory                       >= 1.2        && < 1.4
     , exceptions                      >= 0.7        && < 0.11
+    , fail                            == 4.9.*
```

Comment:
Would you mind keeping the same style as the other version number numbers? Makes it easier to bump the upper bounds in the future. Other than that I'm happy with these changes because I can do a release without bumping a major version. We can talk about the tradeoffs of `mzero` in another PR perhaps, but in general I don't like breaking changes if I can avoid it.

Reply:
Ok

Reply:
@jystic, @HuwCampbell just [forced-pushed](https://github.com/hedgehogqa/haskell-hedgehog/compare/d770d507dc7ab9007a9028786a11e569dbb52049..b400bf14216161e67048f4adbcd71e45bde5e0f6) the change so it's looking good now :+1: